lib/kitchen/driver/oci/blockstorage.rb

```diff
@@ -131,14 +131,25 @@ def attachment_response(attachment_id)
         end
 
         def volume_details(volume)
-          OCI::Core::Models::CreateVolumeDetails.new(
+          details = OCI::Core::Models::CreateVolumeDetails.new(
             compartment_id: oci.compartment,
             availability_domain: config[:availability_domain],
             display_name: volume[:name],
             size_in_gbs: volume[:size_in_gbs],
             vpus_per_gb: volume[:vpus_per_gb] || 10,
             defined_tags: config[:defined_tags]
           )
+          add_volume_source_details(details, volume)
+        end
+
+        def add_volume_source_details(details, volume)
+          if volume.key?(:source_details)
```

Comment:
I'm not a huge fan of adding additional levels to the `volumes` hash.  I think simply providing `volume_id` as a top-level item would be sufficient and it would be a bit clearer to the end user what they are providing.

Reply:
That is fair. I was trying to make it match the terraform resource [here](https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/core_volume.html#source_details) since the API requires both the values `type` (`volume`, `blockVolumeReplica`, and `volumeBackup`)  and `volume_id`.

After further testing though, it's not needed to be specified to get the volume cloning to work through the Ruby SDK - so if we don't care about volume types we can just do a single value at the top level like this:

```yaml
  volumes:
    - name: data
      source_volume_id: 'ocid1.volume.oc1.iad.<ID>' # or volume_id: 'ocid1.volume.oc1.iad.<ID>'
 ```
 
Does that look better? I don't feel like just `volume_id` is clear enough in my opinion, but I'll defer to whatever you choose since it's your repo.

Reply:
I updated the code now, it does as you asked:

```
-----> Creating <default-ubuntu-2204>...
       Waiting for SSH service on 129.213.31.156:22, retrying in 3 seconds
       [SSH] Established
       Creating <data (Clone)>...
       Finished creating <data (Clone)>.
       Attaching <data (Clone)>...
       Finished attaching <data (Clone)>.
       Creating <test-jose-volume-3 (Clone)>...
       Finished creating <test-jose-volume-3 (Clone)>.
       Attaching <test-jose-volume-3 (Clone)>...
       Finished attaching <test-jose-volume-3 (Clone)>.
       Finished creating <default-ubuntu-2204> (1m54.67s).
-----> Test Kitchen is finished. (1m55.77s)
```

Reply:
It's updated now just to use `volume_id` as requested:

```yaml
  volumes:
    - volume_id: 'ocid1.volume.oc1.iad.OCID_1'
    - volume_id: 'ocid1.volume.oc1.iad.OCID_2'
  ```